README.md

```diff
@@ -28,8 +28,140 @@ ember-data-url-templates `>= 0.1.0` is known to work with ember-data `>= 1.0.0-b
 
 More in depth documentation can be found in [the wiki](https://github.com/amiel/ember-data-url-templates/wiki).
 
-### Synopsis
+### Examples
+`urlSegments` are used to build dynamic portions of the url. 
+The appropriate mechanism for passing data from the application to `urlSegments` depends on i) the `DS.store` method used, and ii) where the passed state-data is stored. 
 
+#### snapshot
+```javascript
+// adapters/comment
+
+import Ember from "ember";
+import DS from "ember-data";
+import UrlTemplates from "ember-data-url-templates";
+
+export default DS.RESTAdapter.extend(UrlTemplates, {
+  urlTemplate: '{+host}/comments{/id}',
+  urlForCreateRecord: '{+host}/users/{userId}/comments',
+
+  urlSegments: {
+    userId(type, id, snapshot, query) {
+      return snapshot.belongsTo('user').id;;
```

Comment:
This should be

```javascript
return snapshot.belongsTo('user', { id: true });
```

This way, you can use a relationship's id without the relationship model having been loaded.